### DISCUSSION TITLE
Add a prefix to the generated key in RedisStrategy

### DISCUSSION THREAD

**Discussion by flipee** [fastapi-users/fastapi-users#1042] [Ideas]
This would mitigate attempts to guess valid keys in Redis.

Example:

Imagine a scenario where, for some reason, there is a `ping` key with a value of `1` in the database.

A simple request:

``` shell
curl -X 'GET' \
    http://localhost:8000/api/protected/route' \
    -H 'accept: application/json' \
    -H 'Authorization: Bearer ping'
```
  
Will authorize an attacker on behalf of a valid user (using integer as primary key of course).

Adding a prefix before saving to the database (e.g. `auth_token:[token]`) would mitigate and even if the prefix is known it would be fine since the query would generate a duplicate prefix, e.g. `auth_token:auth_token:[token]`

**Comment by frankie567** [comment]
Hello @flipee 👋 

Thank you for your message, that's indeed an issue I didn't foresee; and it's true that Redis servers are usually used for lot of different things.

Escalating this into an issue!

**Reply by frankie567** [reply]
Issue #1043
